### PR TITLE
Hide/Enable Offer Button to Company Offers Management Section

### LINKED ISSUE #140: Add Hide/Enable Offer Button to Company Offers Management Section
Add a `hide` / `enable` button to the company management offers:

It should be something similar to the review applications page:

image 

The code of the review applications page is available [here](https://github.com/NIAEFEUP/nijobs-fe/blob/develop/src/components/Review/Applications/Actions.js#L25).

You should add the necessary code [here](https://github.com/NIAEFEUP/nijobs-fe/blob/develop/src/components/Company/Offers/Manage/CompanyOffersManagementWidget.js#L66).

You can see examples of how to implement the button functionalities [here](https://github.com/NIAEFEUP/nijobs-fe/blob/develop/src/pages/OfferPage.js#L38) and [here](https://github.com/NIAEFEUP/nijobs-fe/blob/develop/src/components/HomePage/SearchResultsArea/Offer/OfferVisibilityOptions.js).

### PR DESCRIPTION
**Pull request by TiagooGomess**
Video:

https://user-images.githubusercontent.com/50210213/161136722-c8e21a73-f8b0-40b9-a559-cfac970e2acf.mov

Desktop view:

image

Mobile view:

image

Closes #140

### CODE REVIEW

**Comment by TiagooGomess** [comment]
It's ready, I just need to add some tests.

File: src/components/Company/Offers/Manage/CompanyOffersVisibilityActions.js [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+        isDisabled: undefined,
+        isBlocked: undefined,
+    });
+
+    const isHiddenOffer = offer?.isHidden;
+    const offerHiddenReason = offer?.hiddenReason;
+
+    useEffect(() => {
+        setOfferVisibilityState({
+            isDisabled: isHiddenOffer && offerHiddenReason === "ADMIN_REQUEST",
```
Comment (monkin77) [inline src/components/Company/Offers/Manage/CompanyOffersVisibilityActions.js:36]: I believe these values ("ADMIN_REQUEST" and "COMPANY_BLOCKED") could be stored as constants to make it easier to use them in other files as well. Maybe in the offerService?

File: src/components/Company/Offers/Manage/CompanyOffersVisibilityActions.js [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+            addSnackbar({
+                message: visibilityError ? visibilityError : "Unexpected Error. Please try again later.",
+                key: `${Date.now()}-${offer._id}-visibilityError`,
+            });
+        }
+    }, [addSnackbar, offer._id]);
+
+    const handleOfferVisibility = async () => {
+        if (offerVisibilityState.isVisible) {
+            await handleHideOffer({
```
Comment (monkin77) [inline src/components/Company/Offers/Manage/CompanyOffersVisibilityActions.js:87]: Just curious, but is there any reason to await here? or even to define the `handleOfferVisibility` as async?

**Review by monkin77 — COMMENTED** [review]
It's looking great! I just have some small suggestions

File: src/components/Company/Offers/Manage/CompanyOffersVisibilityActions.js (hunk 2 above)
Comment (TiagooGomess) [reply src/components/Company/Offers/Manage/CompanyOffersVisibilityActions.js:87]: Actually, I don't need it, it was a mistake.

File: src/components/Company/Offers/Manage/CompanyOffersVisibilityActions.js (hunk 1 above)
Comment (TiagooGomess) [reply src/components/Company/Offers/Manage/CompanyOffersVisibilityActions.js]: True

**Comment by DoStini** [comment]
What happens if the request takes too long

**Comment by TiagooGomess** [comment]
> What happens if the requests are a bit slow? Shouldn't it disabled while it's fulfilling the request

Yes, it should be disabled. We should implement that also on the offer page buttons. I think it's better to open a new issue for that.

File: src/components/Company/Offers/Manage/CompanyOffersVisibilityActions.js [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+    }, [addSnackbar, offer._id]);
+
+    const handleOfferVisibility = () => {
+        if (offerVisibilityState.isVisible) {
+            handleHideOffer({
+                offer: offer,
+                addSnackbar: addSnackbar,
+                onError: handleOfferVisibilityError,
+            });
+        } else  {
```
Comment (DoStini) [inline src/components/Company/Offers/Manage/CompanyOffersVisibilityActions.js:93]:
```suggestion
        } else {
```

File: src/components/Company/Offers/Manage/CompanyOffersVisibilityActions.js [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+            handleEnableOffer({
+                offer: offer,
+                addSnackbar: addSnackbar,
+                onError: handleOfferVisibilityError,
+            });
+        }
+    };
+
+    return (
+        <Tooltip title={ offerVisibilityState.isVisible ? "Hide Offer" : "Enable Offer"}>
```
Comment (DoStini) [inline src/components/Company/Offers/Manage/CompanyOffersVisibilityActions.js:103]:
```suggestion
        <Tooltip title={offerVisibilityState.isVisible ? "Hide Offer" : "Enable Offer"}>
```

File: src/components/Company/Offers/Manage/CompanyOffersVisibilityActions.js [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+                });
+            })
+            .catch((err) => {
+                if (onError) onError(err);
+            });
+    }, []);
+
+    const handleOfferVisibilityError = useCallback((err) => {
+        if (Array.isArray(err) && err.length > 0) {
+            let visibilityError = null;
```
Comment (DoStini) [inline src/components/Company/Offers/Manage/CompanyOffersVisibilityActions.js:73]: if you dont initialize it, i think it has the same behaviour

**Review by DoStini — CHANGES_REQUESTED** [review]

**Review by monkin77 — APPROVED** [review]
It's looking good!

**Review by DoStini — APPROVED** [review]

**Pull request merged**